src/app/app.routes.ts

```diff
@@ -0,0 +1,27 @@
+import { Routes } from '@angular/router';
+
+import { IndexComponent } from './pages/index.component';
+import { AddCourseComponent } from './pages/addcourse/addcourse.component';
+import { NoContentComponent } from './pages/nocontent/nocontent.component';
+import { LoginPageComponent } from './pages/login-page.component';
+import { CourseListComponent } from './pages/course-list/course-list.component';
+
+
+export const ROUTES: Routes = [
+  { path: '', redirectTo: 'courses', pathMatch: 'full'},
+
+  { path: 'courses', component: IndexComponent,
+    children: [
+        { path: '', component: CourseListComponent },
+        { path: 'new', component: AddCourseComponent },
+        { path: ':id', component: AddCourseComponent }
+      ]
+  },
+  { path: 'courses/:id', component: AddCourseComponent },
+  { path: 'courses/new', component: AddCourseComponent},
```

Comment:
Why do you need `courses/:id` here twice? As a child route for `courses` and as a separate route

Reply:
historical trash, need to remove